superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -271,26 +276,44 @@ function dbReducer(
         [action.payload.name]: action.payload.value,
       };
     case ActionType.parametersChange:
-      if (
-        trimmedState.catalog !== undefined &&
-        action.payload.type?.startsWith('catalog')
-      ) {
-        // Formatting wrapping google sheets table catalog
-        const idx = action.payload.type?.split('-')[1];
-        const catalogToUpdate = trimmedState?.catalog[idx] || {};
-        catalogToUpdate[action.payload.name] = action.payload.value;
-
-        const paramatersCatalog = {};
-        // eslint-disable-next-line array-callback-return
-        trimmedState.catalog?.map((item: CatalogObject) => {
-          paramatersCatalog[item.name] = item.value;
-        });
-
+      if (action.payload.type?.startsWith('catalog')) {
```

Comment:
why have this be a nested loop?

Reply:
Thanks for calling that out. I had originally written logic for the case where `action.payload.type?.startsWith('catalog') && trimmedState.catalog === undefined`, thus the nested logic, but now I see that a form with a catalog will always have a defined catalog state, so I removed the logic and added some comments.